app/views/shipit/ccmenu/project.xml.builder

```diff
@@ -0,0 +1,26 @@
+# Derived from http://timnew.me/blog/2013/04/07/multiple-project-summary-reporting-standard-cctray-xml-feed/
+last_status = 'Unknown'
+if stack.locked?
+  last_status = 'Failure'
+else
+  if deploy.running?
+    last_status = 'pending'
+  elsif deploy.failed?
+    last_status = 'Failure'
+  elsif deploy.error?
+    last_status = 'Exception'
+  elsif deploy.success?
+    last_status = 'Success'
+  end
+end
```

Comment:
Also, in Ruby everything is an expression, so you can do:

``` ruby
last_status = if stack.locked?
  'Failure'
else
  if deploy.running?
    'pending'
  ....
```

or wrap it in a function even:

``` ruby
def last_status
if stack.locked?
  'Failure'
else
  if deploy.running?
    'pending'
  ....
end